lib/dynamoid/transaction/update.rb

```diff
@@ -0,0 +1,68 @@
+require_relative 'action'
+
+module Dynamoid
+  module Transaction
+    class Update < Action
+
+      def run_callbacks
+        (yield; return) unless model
+        model.run_callbacks(:save) do
+          model.run_callbacks(:update) do
+            model.run_callbacks(:validate) do
+              yield
+            end
+          end
+        end
+      end
+
+      def to_h
+        item = attributes.clone
+
+        # ensure types are correct for incoming attributes
+        all_types_string = item.keys.map{|k| [k,{type: :string}]}.to_h
+        item = add_timestamps(item, skip_created_at: true)
+        item = model.attributes.merge(item) if model
+        # TODO only update dirty attributes, deleting the inheritance field will not be needed
+        item.delete(model_class.inheritance_field) # do not update type field
+        item = Dynamoid::Dumping.dump_attributes(item, all_types_string.merge(model_class.attributes))
```

Comment:
What is the purpose of `all_types_string`? To accept not declared fields? It isn't supported in Dynamoid intentionally, `UnknownAttribute` is raised in this case.

Reply:
This was removed.